src/kernel/pm/process.c

```
@@ -236,8 +236,11 @@ void process_yield(void)
  */
 noreturn void process_exit(void)
 {
-    running->state = PROCESS_TERMINATED;
-    process_free(running);
+    if (running != kernel) {
+        running->state = PROCESS_TERMINATED;
+        process_free(running);
+        running = kernel;
+    }
```

Instead of failing silently if the kernel calls this function, could you panic if that is the case?